### PR TITLE
docs: translation Rendering Lists

### PR DESCRIPTION
**Pull request by rccj**
Hi, Please let me know if any problem I need to fix.
Thank you !

### CODE REVIEW

**Comment by rccj** [comment]
Hi @neighborhood999, thank you for your suggestion. It's too many things I have to learning 😂, please take a look again.

File: src/content/learn/rendering-lists.md [hunk 1]
```diff
@@ -1,24 +1,22 @@
...
+你經常會需要用多個相似的 component 來顯示一系列的資料。這個時候你可以在 React 中使用 JavaScript 的 [`filter()`](https://developer.mozilla.org/docs/Web/JavaScript/Reference/Global_Objects/Array/filter) 和 [`map()`](https://developer.mozilla.org/docs/Web/JavaScript/Reference/Global_Objects/Array/map) 來過濾資料並轉換成包含多個 component 的 array 。 
 
 </Intro>
 
 <YouWillLearn>
-
-* How to render components from an array using JavaScript's `map()`
-* How to render only specific components using JavaScript's `filter()`
-* When and why to use React keys
+* 如何使用 JavaScript 的 `map()` 方法處理 array 資料 render component
```
Comment (neighborhood999) [inline src/content/learn/rendering-lists.md:12]: Missing one line 😄.

File: src/content/learn/rendering-lists.md [hunk 2]
```diff
@@ -1,24 +1,22 @@
...
+* 如何使用 JavaScript 的 `filter()` 方法處理 array 資料 render 特定的 component
+* 什麼時候使用及為什麼使用 React 中的 key
 
 </YouWillLearn>
 
-## Rendering data from arrays {/*rendering-data-from-arrays*/}
-
-Say that you have a list of content.
+## 從 array 中 render 資料 {/*rendering-data-from-arrays*/}
+這裡有一個列表
```
Comment (neighborhood999) [inline src/content/learn/rendering-lists.md:19]: Missing one line
```suggestion
假設你有一個內容列表。
```

File: src/content/learn/rendering-lists.md [hunk 3]
```diff
@@ -30,11 +28,12 @@ Say that you have a list of content.
 </ul>
 ```
 
-The only difference among those list items is their contents, their data. You will often need to show several instances of the same component using different data when building interfaces: from lists of comments to galleries of profile images. In these situations, you can store that data in JavaScript objects and arrays and use methods like [`map()`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/map) and [`filter()`](https://developer.mozilla.org/docs/Web/JavaScript/Reference/Global_Objects/Array/filter) to render lists of components from them.
+在這個列表唯一的差異就在於各個內容和資料。在建立介面時，你會經常需要使用不同的資料顯示相同的 component instance：從是評論列表或是個人資料圖片庫。在這些情境下，你可以把需要用到的資料存入 JavaScript 的 object 或 array，並使用 [`map()`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/map) 和 [`filter()`](https://developer.mozilla.org/docs/Web/JavaScript/Reference/Global_Objects/Array/filter) 的方法來 render 一個 component 列表。
```
Comment (neighborhood999) [inline src/content/learn/rendering-lists.md:31]:
```suggestion
在這個列表唯一的差異就在於各個內容和資料。當你建立介面時，常常需要使用不同資料顯示相同的 component，從評論列表到個人資料圖庫。在這些情況下，你可以將資料儲存在 JavaScript 的 object 和array 中，並使用 [`map()`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/map) 和 [`filter()`](https://developer.mozilla.org/docs/Web/JavaScript/Reference/Global_Objects/Array/filter) 的方法來 render 一個 component 列表。
```

File: src/content/learn/rendering-lists.md [hunk 4]
```diff
@@ -30,11 +28,12 @@ Say that you have a list of content.
...
 
-Here’s a short example of how to generate a list of items from an array:
 
-1. **Move** the data into an array:
+這裡有個使用 array 產生一個列表項目的簡單範例：
```
Comment (neighborhood999) [inline src/content/learn/rendering-lists.md:34]: This line at number of **35**.

File: src/content/learn/rendering-lists.md [hunk 5]
```diff
@@ -244,51 +244,51 @@ img { width: 100px; height: 100px; border-radius: 50%; }
...
+  <li>...</li> // 隱式回傳!
 );
 ```
 
-However, **you must write `return` explicitly if your `=>` is followed by a `{` curly brace!**
+然而，**如果你的 `=>` 後面跟著 `{` 大括號，那你必須明確地撰寫 `return` 陳述式！**
 
 ```js
-const listItems = chemists.map(person => { // Curly brace
+const listItems = chemists.map(person => { // 大括號
```
Comment (neighborhood999) [inline src/content/learn/rendering-lists.md:258]:
```suggestion
const listItems = chemists.map(person => { // Curly brace
```

File: src/content/learn/rendering-lists.md [hunk 6]
```diff
@@ -244,51 +244,51 @@ img { width: 100px; height: 100px; border-radius: 50%; }
...
   return <li>...</li>;
 });
 ```
 
-Arrow functions containing `=> {` are said to have a ["block body".](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Functions/Arrow_functions#function_body) They let you write more than a single line of code, but you *have to* write a `return` statement yourself. If you forget it, nothing gets returned!
+包含 `=> {` 的箭頭函式被稱為「區塊主體」。它們允許你撰寫超過一行的程式碼，但你*必須*自己撰寫 `return` 陳述式。如果你忘記了它，則不會回傳任何東西！
```
Comment (neighborhood999) [inline src/content/learn/rendering-lists.md:263]:
```suggestion
包含 `=> {` 的箭頭函式被稱為「區塊主體函式」。它們讓你可以撰寫多行程式碼，但你*必須*自己撰寫 `return` 陳述式。如果你忘了寫，函式就不會回傳任何值！詳細說明可以參考[這裡](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Functions/Arrow_functions#function_body)。
```

**Comment by neighborhood999** [comment]
> Hi @neighborhood999, thank you for your suggestion. It's too many things I have to learn 😂, please take a look again.

Hi @rccj thank you for your contribution. Sorry I left a lot of comments in your PR, most are good to me, but some translation does not correspond to the original sentence and some lines of the number were wrong, so please help me correct them.😄

**Comment by rccj** [comment]
> > Hi @neighborhood999, thank you for your suggestion. It's too many things I have to learn 😂, please take a look again.
> 
> Hi @rccj thank you for your contribution. Sorry I left a lot of comments in your PR, most are good to me, but some translation does not correspond to the original sentence and some lines of the number were wrong, so please help me correct them.😄

Hi @neighborhood999, I appreciate that your suggestion and patience . please take a look again.

File: src/content/learn/rendering-lists.md [hunk 7]
```diff
@@ -1,24 +1,23 @@
...
+* 如何使用 JavaScript 的 `filter()` 方法處理 array 資料 render 特定的 component
+* 什麼時候使用及為什麼使用 React 中的 key
 
 </YouWillLearn>
 
-## Rendering data from arrays {/*rendering-data-from-arrays*/}
-
-Say that you have a list of content.
+## 從 array 中 render 資料 {/*rendering-data-from-arrays*/}
+假設你有一個內容列表。
```
[the remark this answers is not in the document]
Comment (rccj) [reply src/content/learn/rendering-lists.md:21]: @neighborhood999 I add that one line , thank you

**Review by neighborhood999 — APPROVED** [review]
Awesome, let's merge it.

**Pull request merged**